02/intersection.md

```diff
@@ -0,0 +1,348 @@
+# Linked List Intersection
+
+## Problem Statement
+
+Given the heads of two singly linked-lists `head_a` and `head_b`, return the node at which the two lists intersect. If the two linked lists do not intersect, the function should return `None`.
+
+For example, the following two linked lists begin to intersect at the node 8:
+
+![intersecting linked lists example 1](../images/intersection_linked_list_example_1.png)
+
+The following two linked lists do _not_ intersect at all.
+
+![intersecting linked lists example 2](../images/intersection_linked_list_example_2.png)
+
+For this problem, we want to focus on the nodes themselves and not necessarily the *value* inside of the node.
+
+For example, while the following linked lists have tails that share the same values, they are not considered to intersect because the nodes of the linked lists are not the same nodes in memory.
+
+![intersecting linked lists example 3](../images/intersection_linked_list_example_3.png)
+
+There are no cycles anywhere in the linked list structures. Assume any intersection includes the tails of each list.
+
+The linked lists must retain their original structure after the function returns.
+
+<br>
+<details style="max-width: 700px; margin: auto;">
```

Comment:
Is it possible to move this inside the checkpoint? I feel like we don't want to be providing them with a comprehensive set of test cases right before we ask them to write their own?

Reply:
Alternatively, we could ask them to write test cases different than the ones we provide, but I feel like that might get tricky with the edge cases